web/modules/custom/asset_distribution/asset_distribution.module

```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * @file
+ * Contains asset_distribution.module.
+ */
+
+use \Drupal\Core\Entity\EntityInterface;
+use \Drupal\rdf_entity\RdfInterface;
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ *
+ * Handle the reverse reference of the solution-distribution entities.
+ */
+function asset_distribution_rdf_entity_presave(EntityInterface $entity) {
+  if ($entity->bundle() != 'asset_distribution') {
+    return;
+  }
+  // Handle reverse reference in the interoperability solution.
+  /** @var \Drupal\rdf_entity\RdfInterface $rdf_entity */
+  $rdf_entity = \Drupal::routeMatch()->getParameter('rdf_entity');
```

Comment:
No don't rely on the route. This functionality is related to the form that creates the distribution, not to the route that happens to display the form :)

The best way would be to prefill the reference in the form using a hidden value.